addons/account/models/account_bank_statement.py

```diff
@@ -297,6 +297,11 @@ def reconciliation_widget_preprocess(self):
         self.env.cr.execute(sql_query, params)
         st_lines_left = self.env['account.bank.statement.line'].browse([line.get('id') for line in self.env.cr.dictfetchall()])
 
+        if not st_lines_left[0].journal_id.default_credit_account_id or\
+           not st_lines_left[0].journal_id.default_debit_account_id:
+            raise UserError("The journal %s has no default credit or debit account set. Please set this journal its default accounts in its settings."
+                            % st_lines_left[0].journal_id.name)
+
```

Comment:
after discussion with qdp, we would prefer that the code allow to have default_debit/credit_account_id as False and does not crash. 
In this case, the default_debit/credit_account_id are used in order to try and find a partner and assign it automatically, if they are not set, we don't have to try to assign a partner, the user will have to do it itself.

Therefore a better fix would be to add a condition to the `if st_lines_left and stl_to_assign_partner and refs:` in order to enter the conditional statement if we have a default_debit/credit_account_id